ComponentKit/DataSources/CKCollectionViewTransactionalDataSource.mm

```diff
@@ -26,13 +26,14 @@ @interface CKCollectionViewTransactionalDataSource () <
 >
 {
   CKTransactionalComponentDataSource *_componentDataSource;
-  id<CKSupplementaryViewDataSource> _supplementaryViewDataSource;
+  __weak id<CKSupplementaryViewDataSource> _supplementaryViewDataSource;
   CKTransactionalComponentDataSourceState *_currentState;
   CKComponentDataSourceAttachController *_attachController;
 }
 @end
 
 @implementation CKCollectionViewTransactionalDataSource
+@synthesize supplementaryViewDataSource = _supplementaryViewDataSource;
```

Comment:
nit: do you really need to create the ivar given that the property is declared? I think you can remove both the ivar and the synthesize completely.